### PR TITLE
Add section on work synchronization to CONTRIBUTING

### PR DESCRIPTION
**Pull request by shonfeder**
This is a followup to our discussion yesterday. It includes guidance on
my key oversight, which was neglecting to ensure the work was planned.

-------

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -53,6 +57,25 @@ Apalache:
...
+Development on Apalache is distributed. As with any distributed system,
+establishing effective synchronization and consensus on key properties requires
+careful attention.
+
+We synchronize and coordinate our work using GitHub issues. When an issue is
+assigned to someone (or to several people), it serves as a token indicating that
+substantive work on the item described on the issue is being done (or will be
+done) by the assigned party. To avoid race conditions, we don't work on
+substantive matters without first holding such a token. If you are eager to work
+on something to which another party is assigned, please coordinate with them. It
```
Comment (konnov) [inline CONTRIBUTING.md:71]:
```suggestion
on something to which another party is assigned, please coordinate with them, e.g., by commenting on the related issue. It
```

**Review by konnov — APPROVED** [review]
Looks great! Just one comment. Thanks for taking care of this!

**Comment by shonfeder** [comment]
Thanks for the review!

**Review by Kukovec — APPROVED** [review]
LGTM

**Pull request merged**